modules/apps/headless/headless-object/headless-object-impl/src/main/java/com/liferay/headless/object/internal/dto/v1_0/converter/CollaboratorDTOConverter.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.object.internal.dto.v1_0.converter;
+
+import com.liferay.headless.delivery.dto.v1_0.util.CreatorUtil;
+import com.liferay.headless.object.dto.v1_0.Collaborator;
+import com.liferay.petra.function.transform.TransformUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.model.UserGroup;
+import com.liferay.portal.kernel.service.UserGroupLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterContext;
+import com.liferay.sharing.model.SharingEntry;
+import com.liferay.sharing.security.permission.SharingEntryAction;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mikel Lorza
+ */
+@Component(
+	property = "dto.class.name=com.liferay.object.model.ObjectEntryFolder",
+	service = DTOConverter.class
+)
+public class CollaboratorDTOConverter
+	implements DTOConverter<SharingEntry, Collaborator> {
+
+	@Override
+	public String getContentType() {
+		return Collaborator.class.getSimpleName();
+	}
+
+	@Override
+	public Collaborator toDTO(
+			DTOConverterContext dtoConverterContext, SharingEntry sharingEntry)
+		throws Exception {
+
+		User user;
+		UserGroup userGroup;
+
+		if (sharingEntry.getToUserId() > 0) {
```

Comment:
@Mikellorza, wouldn't it be better to have a type to differentiate between toUserId and toUserGroupId? So as not depend of the the values of the properties? 

I think we have a property that's oneOf, so we could distinguish between a UserCollaborator and a UserGroupCollaborator so as not to mix both

Reply:
we do not have that UserCollaborator and UserGroupCollaborator, so, maybe I am missing something

Reply:
Forget my comment, but we can use the enum that we talked about, like

```
viewableBy:
                    description:
                        A write-only property that specifies the document’s default permissions.
                    enum: [Anyone, Members, Owner]
                    type: string
```